WalletWasabi.Tests/UnitTests/StartWasabiOnSystemStartupTests.cs

```diff
@@ -1,31 +1,64 @@
-using System;
+using System.IO;
 using System.Runtime.InteropServices;
 using System.Threading.Tasks;
 using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Gui;
+using WalletWasabi.Helpers;
+using WalletWasabi.Tests.Helpers;
 using Xunit;
 
 namespace WalletWasabi.Tests.UnitTests
 {
 	public class StartWasabiOnSystemStartupTests
 	{
+		private readonly WindowsStartupHelper _windowsHelper = new();
+
+		private readonly MacOsStartupHelper _macOsHelper = new();
+
 		[Fact]
 		public async Task ModifyStartupOnDifferentSystemsTestAsync()
 		{
+			UiConfig originalConfig = GetUiConfig();
+
 			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
 			{
 				await StartupHelper.ModifyStartupSettingAsync(true);
+
+				Assert.True(_windowsHelper.RegistryKeyExists());
+
 				await StartupHelper.ModifyStartupSettingAsync(false);
+
+				Assert.False(_windowsHelper.RegistryKeyExists());
 			}
 			else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
 			{
-				await Assert.ThrowsAsync<NotImplementedException>(async () => await StartupHelper.ModifyStartupSettingAsync(true));
-				await Assert.ThrowsAsync<NotImplementedException>(async () => await StartupHelper.ModifyStartupSettingAsync(false));
+				await StartupHelper.ModifyStartupSettingAsync(true);
+
+				Assert.True(File.Exists(LinuxStartupHelper.FilePath));
+				Assert.Equal(LinuxStartupHelper.ExpectedDesktopFileContent, LinuxStartupHelper.GetFileContent());
+
+				await StartupHelper.ModifyStartupSettingAsync(false);
+
+				Assert.False(File.Exists(LinuxStartupHelper.FilePath));
 			}
 			else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
 			{
 				await StartupHelper.ModifyStartupSettingAsync(true);
+
```

Comment:
Add a comment here why the value is not read back.

Reply:
Done.